packages/clerk-js/src/ui/components/SignIn/utils.ts

```diff
@@ -134,3 +135,8 @@ export function determineStartingSignInSecondFactor(secondFactors: SignInFactor[
 const resetPasswordStrategies: SignInStrategy[] = ['reset_password_phone_code', 'reset_password_email_code'];
 export const isResetPasswordStrategy = (strategy: SignInStrategy | string | null | undefined) =>
   !!strategy && resetPasswordStrategies.includes(strategy as SignInStrategy);
+
+export function navigateWithError(clerk: Clerk, navigate: (to: string) => Promise<unknown>, to: string, err) {
```

Comment:
I'd add this method as an internal method on Clerk Singleton. It feels like it's a better home with the current architecture.